NIFI-4895: Added backend code to give users the ability to forcibly t…

markap14 opened the pull request:
…erminate a processor if the processor does not complete after clicking Stop

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

mcgilman commented:
Will review...

mcgilman commented on nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/scheduling/StandardProcessScheduler.java:
```diff
@@ -138,7 +138,6 @@ public void setMaxThreadCount(final SchedulingStrategy schedulingStrategy, final
             return;
         }
 
-        agent.setMaxThreadCount(maxThreadCount);
```
Was this intended? It appears this method `setMaxThreadCount` no longer does anything. Assuming it was intended, we can likely remove the rest of the code in this method. If the method is no longer relevant, should it be deprecated?

mcgilman commented on nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractPort.java:
```diff
@@ -497,6 +497,12 @@ public void yield() {
     }
 
+    @Override
+    public void yield(final long yieldDuration, final TimeUnit timeUnit) {
```
Should the other `yield` method invoke this one?

markap14 replied:
Yes. :)

markap14 commented on nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessorResource.java:
```diff
@@ -209,6 +211,50 @@ public Response getProcessor(
+            return replicate(HttpMethod.POST);
+        }
+
+        final Revision requestRevision = new Revision(version == null ? null : version.getLong(), clientId.getClientId(), id);
```
Good call.

markap14 commented:
@mcgilman thanks for the review! I pushed another commit that addresses all of the issues that you raised above. I also did another code review myself and found a couple of minor issues that I also addressed. Thanks!

markap14 commented:
@mcgilman sorry - i had forgotten to push the branch to remote! Updated commit is now available.
